apel/common/datetime_utils.py

```diff
@@ -58,14 +58,19 @@ def parse_timestamp(datetime_string):
 
 
 def parse_time(timestring):
-    '''
-    Return seconds from times of the form d-h:m:s or h:m:s.
-    '''
+    """Return integer seconds from times of form d-h:m:s, h:m:s or m:s.s."""
     if '-' in timestring:
         days, sub_days = timestring.split('-')
     else:
         days, sub_days = 0, timestring
-    hours, minutes, seconds = sub_days.split(':')
+
+    if '.' in sub_days:
```

Comment:
Maybe I just like regexps too much, but I'd probably try to parse the times with something like:
`(?:(?P<days>\d)-)?(?:(?P<hours>\d{2}):)?(?P<minutes>\d{2}):(?P<seconds>\d{2}(?:\.\d+)?)`

Although... maybe you could loop over a series of `strptime()` formats, but there would be problems with day numbering and fractional seconds... nope, regexps FTW.

Reply:
Feel free to submit a PR! :wink: 

It's a fair point, but I wanted to fix the current problem rather than introduce a new factor in.